src/definition/populate_test.go

```diff
@@ -734,6 +734,38 @@ func TestIntegrationPopulator_PrometheusFormatFilterNamespace(t *testing.T) {
 	}
 }
 
+func TestIntegrationPopulator_CustomMsTypeGuesser(t *testing.T) { //nolint: paralleltest
+	intgr, err := integration.New("nr.test", "1.0.0", integration.InMemoryStore())
+	require.NoError(t, err)
+
+	customMsTypeGuesser := func(_, _, _ string, _ definition.RawGroups) (string, error) {
+		return "Custom", nil
+	}
+
+	config := testConfig(intgr)
+	config.Specs = definition.SpecGroups{
+		"test": definition.SpecGroup{
+			TypeGenerator:   fromGroupEntityTypeGuessFunc,
+			NamespaceGetter: kubeletMetric.FromLabelGetNamespace,
+			MsTypeGuesser:   customMsTypeGuesser,
+			Specs: []definition.Spec{
+				{"metric_1", definition.FromRaw("raw_metric_name_1"), metric.GAUGE, false},
+			},
+		},
+	}
+
+	populated, errs := definition.IntegrationPopulator(config)
+	assert.True(t, populated)
+	assert.Empty(t, errs)
+	for _, e := range intgr.Entities {
+		if e.Metadata.Namespace != "k8s:cluster" { // the custom MsTypeGuesser does not apply to the cluster entity.
+			for _, ms := range e.Metrics {
+				assert.Equal(t, "Custom", ms.Metrics["event_type"])
+			}
+		}
```

Comment:
NIT: I like avoiding nested code in cases like this as IMHO it makes it easier to follow the code.
```suggestion
		if e.Metadata.Namespace == "k8s:cluster" { // the custom MsTypeGuesser does not apply to the cluster entity.
		    continue
		}
		
		for _, ms := range e.Metrics {
		    assert.Equal(t, "Custom", ms.Metrics["event_type"])
		}
```

Reply:
Thanks! Done: [c423d36](https://github.com/newrelic/nri-kubernetes/pull/577/commits/c423d360b45742a1d20edf100aab0043ca2b87d3)